pkg/ssh/ssh_test.go

```diff
@@ -328,3 +328,39 @@ func TestSSHUser(t *testing.T) {
 	}
 
 }
+
+type slowDialer struct {
+	delay time.Duration
+	err   error
+}
+
+func (s *slowDialer) Dial(network, addr string, config *ssh.ClientConfig) (*ssh.Client, error) {
+	time.Sleep(s.delay)
+	if s.err != nil {
+		return nil, s.err
+	}
+	return &ssh.Client{}, nil
+}
+
+func TestTimeoutDialer(t *testing.T) {
+	testCases := []struct {
+		delay             time.Duration
+		timeout           time.Duration
+		err               error
+		expectedErrString string
+	}{
+		// delay > timeout should cause ssh.Dial to timeout.
+		{1 * time.Second, 0, nil, "Timed out dialing"},
+		// delay < timeout should return the result of the call to the dialer.
+		{0, 1 * time.Second, nil, ""},
+		{0, 1 * time.Second, fmt.Errorf("Test dial error"), "Test dial error"},
+	}
+	for _, tc := range testCases {
+		dialer := &timeoutDialer{&slowDialer{tc.delay, tc.err}, tc.timeout}
+		_, err := dialer.Dial("tcp", "127.0.0.1:8080", &ssh.ClientConfig{})
```

Comment:
Will this test fail if it's run on a machine that has something listening on port 8080?